lua/oil/util.lua

```diff
@@ -930,19 +930,26 @@ M.get_icon_provider = function()
   local _, mini_icons = pcall(require, "mini.icons")
   ---@diagnostic disable-next-line: undefined-field
   if _G.MiniIcons then -- `_G.MiniIcons` is a better check to see if the module is setup
-    return function(type, name)
-      return mini_icons.get(type == "directory" and "directory" or "file", name)
+    return function(type, name, ft)
+      local icon, hl, is_default = mini_icons.get(type == "directory" and "directory" or "file", name)
+      if icon == nil or is_default then
+        icon, hl = mini_icons.get("filetype", ft or "")
+      end
```

Comment:
Probably should only do this when `ft` is non-nil, right?

Reply:
Actually, I think for both devicons and MiniIcons it makes sense to use the filetype _first_ if it's provided, as that implementation should both be faster and more correct, and only use the original logic if the filetype is nil.

Reply:
I have pushed some commits to address this, mini.icons changed, but devicons actually was simplified, since it handles the `ft == nil` logic already.